### PR TITLE
written unit-tests for login_signup view in django

### PR DESCRIPTION
**Pull request by subhangi2731**
closes issue #76 
Wrote unit tests for login_signup view in the accounts app. The tests are added in a file test_views in test directory of accounts app and tested it written in django

### CODE REVIEW

**Comment by subhangi2731** [comment]
@garg3133 please review and merge my pr. this issue was very tough and thanks for helping out. It literally took me 2 weeks to write the code and test it

**Comment by subhangi2731** [comment]
@garg3133 please merge my pr

File: accounts/tests/test_views.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+            street_address1 = 'awdwdawd',
+            street_address2 = 'awdawdawd',
+            contact_no = '9999999999',
+            city = 'Montreal',
+            state = 'Quebec',
+            pincode = 98743
+        )
+
+    def test_view_url_exists_at_desired_location(self):
+        response = self.client.get('')
```
Comment (garg3133) [inline accounts/tests/test_views.py:34]: You need to check for the `/accounts/login/` link.

**Review by garg3133 — COMMENTED** [review]
@subhangi2731 Most of the test cases looks really great, you've written them nicely. However, few of the test cases seems wrong, as reviewed below. Please check and correct them.

There's a few functionalities left from being tested in this PR, like error when `auth=False`, redirection to complete profile page if `is_active=True` and there is no user profile, etc. But you can do that in a separate Level2 PR if you want (I'll label this PR as Level3).

Basically, we need to test all the code paths in the login_signup view and all the possible results.

**Comment by garg3133** [comment]
@subhangi2731 Let me know if you face any problem or is stuck anywhere.

**Comment by subhangi2731** [comment]
@garg3133 please merge this i will create a new another issue with more tests as u said earlier in this pr and u will give level 2 to that new issue related to this tests

**Comment by garg3133** [comment]
@subhangi2731 But some of these tests are wrong, and I cannot merge them until these are fixed.

You just need to fix the three things which I pointed out above, and you'll be good to go. These are very small fixes anyway and won't take much time at your end.

**Comment by subhangi2731** [comment]
@garg3133 def test_view_login_with_correct_data(self):
        response = self.client.post(reverse('accounts:login_signup'), {
            'email' : 'testuser2@gmail.com',
            'password' : '2HJ1vRV0Z&3iD',
            'submit' : 'login'
        })
        # Check it redirects correctly to login_signup
        self.assertEqual(response.status_code, 302)
        self.assertRedirects(response, 'home:dashboard')

I am getting this error please help

apps.volunteers.models.Volunteer.DoesNotExist: Volunteer matching query does not exist.

**Comment by garg3133** [comment]
@subhangi2731 It seems like you also need to create a `Volunteer` model instance along with `Profile` instance. That should be a quick thing to do as you just need to add 5 fields to Volunteer instance, `profile`, `roll_no`, `dob`, `batch`, `programme`.

**Comment by subhangi2731** [comment]
@garg3133 now please merge my pr I did all the changes and everything as u have said

**Comment by subhangi2731** [comment]
@garg3133 please merge my pr I have done everything

**Comment by garg3133** [comment]
This looks perfect now. Merged!

Thanks, @subhangi2731 for your contribution 🚀
